### DISCUSSION TITLE
Water marking on SVG files failing in Linux

### DISCUSSION THREAD

**Discussion by sripvk1234** [ImageMagick/ImageMagick#4257] [General]
I'm running below command on SVG files to apply water marking
convert -size 240x240 xc:none -font C059-Bold -fill grey -gravity NorthWest -draw "text 10,10 'Copyright'" -gravity SouthEast -draw "text 5,15 'Copyright'" miff:- | composite -tile - abc.svg abcd.svg

It throws below given error in Linux environment
composite: delegate failed `'potrace' --svg --output '%o' '%i'' @ error/delegate.c/InvokeDelegate/1913.

In Windows environment, the command is successful. Here is the command
magick -size 240x240 xc:none -fill grey -gravity NorthWest -draw "text 10,10 'Copyright'" -gravity SouthEast -draw "text 5,15 'Copyright'" miff:- | magick composite -tile - abc.svg abcd.svg

ImageMagick version details
Version: ImageMagick 7.1.0-4 Q16 x86_64 2021-07-18 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC

I have tried same command using ImageMagick 7.0.10-62 Q16 x86_64 2021-02-07 and did not see above mentioned error. But the output content is coming as

```
<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
  "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg width="600" height="976">
<g style="fill:black;fill-opacity:1;stroke:none;stroke-width:1;stroke-opacity:1;fill-rule:nonzero;</svg>
```

**Comment by snibgo** [comment]
ImageMagick is a raster image processor. That means it deals with pixels, not vectors. SVG is a vector image format.

Your command will rasterize the input SVG, and draw some rasterized text on that image. Finally it converts the raster image to an output SVG.

In that final stage, your two installations are trying to do different things. One of them is trying to run potrace to trace the raster image to convert it to vectors. I don't know what the other does. Perhaps it succeeds in calling potrace.

I don't know what result you want. Does the "convert" command do what you want? If it does, then insert "-verbose" in the second half to see if it calls potrace. Or perhaps the result SVG contains many small circles, each containing one of the raster pixels.

Perhaps the result you want is the input vector SVG with added elements that are your text as vector elements. IM can't do that as it is a raster processor. IF you want that, then I suggest you edit the SVG directly, with text processing tools.

It might help if you show a sample input SVG.

**Comment by sripvk1234** [comment]
Attached Sample SVG file
[scuba-diving-147683.zip](https://github.com/ImageMagick/ImageMagick/files/7216649/scuba-diving-147683.zip)

**Comment by snibgo** [comment]
Please say what result you want. This might be:

1. A raster image. If so, do you want this in a SVG wrapper? Why?

2. A vector SVG file. If so, then ImageMagick is the wrong tool.

3. Something else.
